### PR TITLE
Automate User 'Rider' Role Toggling

### LINKED ISSUE #9: FEATURE: Accepting a rider application should modify the user's role
### As a...

admin

### I can...

accept rider applications that will automatically assign the user to be a rider

### So that...

I dont have to manually change their roles in the user table

### Discussion

Right now, when a ride application is accepted, only the status of the ride application is changed. So the admin has to go to the table of users and find the user who submitted the application and manually toggle their role.

What is better is, if the application is accepted, then the user who submitted the application gains the role automatically.

### Acceptance Criteria

- [ ] When an admin accepts a rider application through the review form, the user gains the `Rider` role.

### Implementation Todos

Frontend:
- [ ] In `RiderApplicationEditForm.js` modify the `handleApprove` method to include toggling of the applicant's role. You can look at `UsersTable.js` to see how it is done there and adapt appropriately.

### PR DESCRIPTION
**Pull request by gracefeng05**
In this PR, I modified `RiderApplicationEditForm.js` to automatically toggle user’s ‘Rider’ role when rider application is approved. I did this according to the example in `UsersTable.js`.

This removes the need for admins to manually toggle ‘Rider’ role for users.

https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-7/assets/114462302/13c72b70-6982-4b96-acd9-0c25afbd9dde

Closes #9

### CODE REVIEW

**Review by katelarrick — APPROVED** [review]
I see the modified 'handleApprove' method, and it looks like it includes toggling of the applicant's role.

File: frontend/src/main/components/RiderApplication/RiderApplicationEditForm.js [hunk 1]
```diff
@@ -1,54 +1,75 @@
...
     const {
         register,
         formState: { errors },
         handleSubmit,
         getValues
-    } = useForm(
-        { defaultValues: initialContents }
-    );
+    } = useForm({ defaultValues: initialContents });
     // Stryker enable all
```
Comment (pconrad) [inline frontend/src/main/components/RiderApplication/RiderApplicationEditForm.js:17]: `// Stryker enable all` should be changed to `// Stryker restore all`.

`// Stryker enable all` doesn't actually do anything, so all of the Stryker checking is disabled after this line.

File: frontend/src/main/components/RiderApplication/RiderApplicationEditForm.js [hunk 2]
```diff
@@ -1,54 +1,75 @@
...
+        };
+    }
+
+    // Stryker disable all
+    const toggleRiderMutation = useBackendMutation(
+        cellToAxiosParamsToggleRider,
+        {},
+        ["/api/admin/users"]
+    );
+    // Stryker enable all
```
Comment (pconrad) [inline frontend/src/main/components/RiderApplication/RiderApplicationEditForm.js:41]:
```suggestion
    // Stryker restore all
```

File: frontend/src/tests/components/RiderApplication/RiderApplicationEditForm.test.js [hunk 3]
```diff
@@ -19,10 +19,20 @@ describe("RiderApplicationEditForm tests", () => {
...
+                </Router>
+            </QueryClientProvider>
+        );
+    };
+
+    test('handleAction submits data and navigates when approving', async () => {
         const mockSubmitAction = jest.fn();
     
-        render(
+        renderWithProviders(
```
Comment (pconrad) [inline frontend/src/tests/components/RiderApplication/RiderApplicationEditForm.test.js:35]: Nice!  I like that you factored out the providers.

**Review by pconrad — CHANGES_REQUESTED** [review]
A few places where we have `// Stryker enable all` which, as it turns out, isn't a thing.

We need to replace that with `// Stryker restore all` everywhere that it appears.

File: frontend/src/main/components/RiderApplication/RiderApplicationEditForm.js (hunk 1 above)
Comment (gracefeng05) [reply frontend/src/main/components/RiderApplication/RiderApplicationEditForm.js:17]: In UsersTable.js, which I didn't touch, there are multiple // Stryker enable all comments. Are these mistakes in the code? I imagine if they were to be replaced by // Stryker restore all, many mutations would survive UsersTable.test.js.

Screenshot 2024-05-22 at 5 34 44 PM

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**